osp/core/ontology/namespace_registry.py

```diff
@@ -109,9 +109,18 @@ def _get(self, name):
                                      iri=self._namespaces[name])
         raise KeyError("Namespace %s not installed." % name)
 
-    def update_namespaces(self, modules=[]):
-        """Update the namespaces of the namespace registry."""
+    def update_namespaces(self,
+                          modules=None,
+                          remove=None):
+        """Update the namespaces of the namespace registry.
+
+        Only needed for Python 3.6.
+        """
+        modules = list(modules) if modules is not None else []
+        remove = list(remove) if remove is not None else []
```

Comment:
You could maybe simplify this by setting the default values as `[]` directly?
Also FYI, I believe converting a set to a list and then iterating is less efficient than iterating the set directly:
```py
In [1]: import timeit

In [2]:  a = [x * x for x in range(1, 400)]

In [3]: s = set(a)

In [4]: timeit.timeit("for i in s: pass", "from __main__ import s")
Out[4]: 5.616416699999718

In [5]: timeit.timeit("l = list(s)\nfor i in l: pass", "from __main__ import s")
Out[5]: 6.821328500000163
```

Reply:
In general, using a mutable default argument in Python is bad practice, because if the argument is modified inside the function (or passed to any other function and then modified), then the _default_ is also modified. On the next function call, the list would be no longer empty by default. It is true that whether this will cause problems or not depends on the specific code, in the function, but again, in general if you do this then you have to be very vigilant about what you do, so the best is to avoid mutable arguments.

About the conversion, true. In fact, I do not even need a list, nor a set, so I should not be converting anything. Just an iterable works. I will change this.